web/src/app/projects/create-edit/create-edit.component.ts

```diff
@@ -67,4 +69,11 @@ export class CreateEditProjectComponent implements OnInit {
     this.createEditSubscription.unsubscribe();
     this.projectSubscription.unsubscribe();
   }
+
+  /**
+   * Getter for url
+   */
+  get url(): AbstractControl {
+    return this.projectForm.get('url');
+  }
```

Comment:
It doesn't look like this is used anymore? 

`public projectForm: FormGroup;` is public and used in the html `<mat-error *ngIf="projectForm.get(['url']).hasError('pattern')">`

Reply:
I have created this method to get the url control and set its value after stripe of the trailing slash

Reply:
I think we should always have getters for all properties of form instead of using formgroup everywhere to find that field.
Suppose if we have to check 3 or 4 validation error then we have to write four time 
projectForm.get(['url'])

Reply:
moved the stripe call in service from component.